apps/storybook/src/stories/transactions/TransactionTable.tsx

```diff
@@ -0,0 +1,217 @@
+import { Button } from "@/components/ui/button";
+import {
+	Table,
+	TableBody,
+	TableCell,
+	TableHead,
+	TableHeader,
+	TableRow,
+} from "@/components/ui/table";
+import { TransactionMeta } from "@/lib/domain/transaction/transaction";
+import { getShortHex } from "@/utils/hex";
+import {
+	ArrowUpDown,
+	ChevronLeft,
+	ChevronRight,
+	ExternalLink,
+} from "lucide-react";
+import React, { useState } from "react";
+import { formatEther, formatGwei } from "viem";
+import { AddressBadge } from "../identity/AddressBadge";
+
+export interface TransactionTableProps {
+	transactions: TransactionMeta[];
+	itemsPerPage?: number;
+	withPagination?: boolean;
+}
+
+export function TransactionTable({
+	transactions,
+	itemsPerPage = 10,
+	withPagination = false,
+}: TransactionTableProps) {
+	const [currentPage, setCurrentPage] = useState(1);
+	const [sortConfig, setSortConfig] = useState<{
+		key: keyof TransactionMeta;
+		direction: "asc" | "desc";
+	} | null>(null);
+
+	const indexOfLastItem = currentPage * itemsPerPage;
+	const indexOfFirstItem = indexOfLastItem - itemsPerPage;
+
+	const sortedTransactions = React.useMemo(() => {
+		let sortableTransactions = [...transactions];
+		if (sortConfig !== null) {
+			sortableTransactions.sort((a, b) => {
+				if (
+					(a[sortConfig.key || "transactionIndex"] || 0) <
+					(b[sortConfig.key || "transactionIndex"] || 0)
+				) {
+					return sortConfig.direction === "asc" ? -1 : 1;
+				}
+				if (
+					(a[sortConfig.key || "transactionIndex"] || 0) >
+					(b[sortConfig.key || "transactionIndex"] || 0)
+				) {
+					return sortConfig.direction === "asc" ? 1 : -1;
+				}
+				return 0;
+			});
+		}
+		return sortableTransactions;
+	}, [transactions, sortConfig]);
+
+	const currentTransactions = sortedTransactions.slice(
+		indexOfFirstItem,
+		indexOfLastItem,
+	);
+
+	const totalPages = Math.ceil(transactions.length / itemsPerPage);
+
+	const requestSort = (key: keyof TransactionMeta) => {
+		let direction: "asc" | "desc" = "asc";
+		if (
```

Comment:
if it's a toggle, we could use direction as `isAsc`  flag and just set isAsc = !isAsc?

separate flag for isSorted if we don't always want to sort